src/controller/CHIPDeviceController.cpp

```diff
@@ -880,6 +880,20 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, RendezvousParam
     VerifyOrExit(mDeviceBeingPaired < kNumMaxActiveDevices, err = CHIP_ERROR_NO_MEMORY);
     device = &mActiveDevices[mDeviceBeingPaired];
 
+    // If the CSRNonce is passed in, using that else using a random one..
+    if (params.HasCSRNonce())
+    {
+        device->SetCSRNonce(params.GetCSRNonce());
```

Comment:
Right here, since we know GetCSRNonce has a value, we'd do `SetCSRNonce(params.GetCSRNonce().Value());` and not have the callee take an `Optional` that it then proceeds to assume has a value.

Reply:
Done.